src/hardware.py

```diff
@@ -66,3 +77,50 @@ def get_bmc_address() -> t.Optional[str]:
     except subprocess.CalledProcessError:
         logger.debug("IPMI is not available")
     return None
+
+
+def _split_at(s: str, c: str, n: int) -> t.Tuple[str, str]:
+    """Split a string 's' at the 'n'th occurrence of delimiter 'c'.
+
+    Parameters:
+        s (str): The string to split.
+        c (str): Delimiter for splitting.
+        n (int): Occurrence of 'c' to split at.
+
+    Returns:
+        Tuple[str, str]: The string before and after the 'n'th occurrence of 'c'.
+
+    Example:
+        >>> _split_at('one,two,three,four,five', ',', 2)
+        ('one,two', 'three,four,five')
+
+        This will split the string at the 2nd occurrence of ',', resulting in two parts:
+        'one,two' and 'three,four,five'.
+    """
+    words = s.split(c)
+    return c.join(words[:n]), c.join(words[n:])
+
+
+def hwinfo(*args: str) -> t.Dict[str, str]:
+    """Run hwinfo command and return output as dictionary.
+
+    Args:
+        args: Probe for a particular hardware class.
+    Returns:
+        hw_info: hardware information dicturary
+    """
+    apt.add_package("hwinfo", update_cache=False)
+    hw_classes = list(args)
+    for idx, hw_item in enumerate(args):
+        hw_classes[idx] = "--" + hw_item
+    hw_classes.insert(0, "hwinfo")
+
+    output = subprocess.check_output(hw_classes, text=True)
```

Comment:
```suggestion
    hw_info_cmd = ["hwinfo"] + hw_classes

    output = subprocess.check_output(hw_info_cmd, text=True)
```
It feel more readable to define a separate `hw_info_cmd` variable that would be run by `subprocess.check_output`.